### DISCUSSION TITLE
How do I override and extend a partial without copying it?

### DISCUSSION THREAD

**Discussion by samuelcolvin** [squidfunk/mkdocs-material#3917] [Q&A] [answered]
I want to add content to the end of `partials/nav.html` - is there an existing way to achieve this without copying the default content from the file's default content?

I've include `{% include 'partials/nav.html' %}` in my `overrides/partials/nav.html` I get a recursion error.

**Comment by squidfunk** [comment, marked as answer]
Thanks for asking. Good question, I don't think it's possible, as partials are, as far as I understand, overloaded by MkDocs, not Jinja, i.e. the directory structures of the theme and your overrides are merged. Thus, there's no access to the parent partial from the override. However, this is just my guess, so it could be a good idea to ask the maintainers of MkDocs.

**Reply by samuelcolvin** [reply]
thanks so much. Good to know I'm not missing anything obvious.

**Reply by squidfunk** [reply]
Another option might be to override the `site_nav` block and add your changes after the inclusion of the nav partial:

https://github.com/squidfunk/mkdocs-material/blob/770b1b9084c0f090c0a6349b554e5af100a01885/src/base.html#L250-L290

History shows that those lines are very rarely changed. The navigation itself does change more often.
